nominal/core.py

```diff
@@ -802,6 +1032,35 @@ def get_attachments(self, rids: Iterable[str]) -> Sequence[Attachment]:
         return list(self._iter_get_attachments(rids))
 
 
+def _create_checklist_builder_from_yaml(checklist_config_path: str, client: NominalClient) -> ChecklistBuilder:
+    with open(checklist_config_path, "r") as file:
+        checklist_dict = yaml.safe_load(file)
+        data = checklist_dict["checklist"]
+        
+        checklist_builder = ChecklistBuilder(
+            name=data["name"],
+            assignee_email=data["assignee_email"],
+            description=data.get("description"),
+            default_ref_name=data.get("default_ref_name"),
+        ).add_metadata(
+            properties=data.get("properties"),
+            labels=data.get("labels"),
+        )
+        checklist_builder._client = client
+
+        if "variables" in data:
+            for variable_dict in data["variables"]:
+                variable_dict["expression"] = _remove_newlines(variable_dict["expression"])
+                checklist_builder.add_checklist_variable(**variable_dict)
+
+        if "checks" in data:
+            for check_dict in data["checks"]:
+                check_dict["expression"] = _remove_newlines(check_dict["expression"])
+                checklist_builder.add_check(**check_dict)
```

Comment:
I just inlined this and went straight from dict into the args of the method - the CreateVar / CreateCheck objects themselves extend the pydantic object